### PR TITLE
Add test for `internal/compress/gob.go`

### PR DESCRIPTION
**Pull request by hlts2**
### Description:

I added test case for `internal/compress/gob.go` and I refactored based on [code guideline](https://github.com/vdaas/vald/blob/master/docs/contributing/coding-style.md#error-checking).

### Environment:

- Go Version: 1.14.4
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.0

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/compress/gob_test.go [hunk 1]
```diff
@@ -92,31 +54,32 @@ func TestNewGob(t *testing.T) {
...
-		           args: args {
-		           opts: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "returns (Compressor, nil) when option is empty",
```
Comment (kevindiu) [inline internal/compress/gob_test.go:58]: please create `returns (Compressor, nil) when option is not empty` test case :)

File: internal/compress/gob.go [hunk 2]
```diff
@@ -83,6 +89,8 @@ type gobReader struct {
 	decoder gob.Decoder
 }
 
+// Read returns the number of bytes for read p (0 <= n <= len(p)).
+// If any errors occurs, it will return an error.
 func (gr *gobReader) Read(p []byte) (n int, err error) {
 	err = gr.decoder.Decode(&p)
 	if err != nil {
```
Comment (kevindiu) [inline internal/compress/gob.go:96]:
```suggestion
	if err = gr.decoder.Decode(&p); err != nil {
```

File: internal/compress/gob.go [hunk 3]
```diff
@@ -101,6 +110,8 @@ type gobWriter struct {
 	encoder gob.Encoder
 }
 
+// write returns the number of bytes written from p (0 <= n <= len(p)).
+// if any errors occurs, it will return an error.
 func (gw *gobWriter) Write(p []byte) (n int, err error) {
 	err = gw.encoder.Encode(&p)
 	if err != nil {
```
Comment (kevindiu) [inline internal/compress/gob.go:117]:
```suggestion
	if err = gw.encoder.Encode(&p); err != nil {
```

File: internal/compress/gob_test.go (hunk 1 above)
Comment (hlts2) [reply internal/compress/gob_test.go:58]: sorry. I already created this one.
https://github.com/vdaas/vald/blob/test/internal/compress/gob/internal/compress/gob_test.go#L72-L75

and gob has no option.

**Review by kevindiu — APPROVED** [review]
LGTM

**Review by vankichi — APPROVED** [review]
LGTM

**Pull request merged**